### ISSUE TITLE
Could we have an option to add datalist to ScaleTextField.

### ISSUE THREAD

**Issue by gaurishkamat** [telekom/scale#447]
Could we have an option to add datalist to the text field. 
For eg: 
```
<input type="text" placeholder="Enter some text" list="my-list"></input>
<datalist id="my-list">
	<option value="Apple">
	<option value="Banana">
	<option value="Orange">
</datalist>
```

Also any way to turn-off the autocomplete on these text field.

Greetings!!

**Comment by acstll** [comment]
Yes, we definitely can :)

It would be awesome if you could contribute a PR for this, is that something you'd be willing to do? (Of course, it's alright to say "no")

Thanks for opening this @gaurishkamat

**Comment by gaurishkamat** [comment]
I would definitely love to work on this.

**Comment by acstll** [comment]
That's great news. Let us know should you need any help. Thank you! @gaurishkamat

**Comment by gaurishkamat** [comment]
Hi @acstll, How do I test my changes before adding a PR? Are there any steps to be followed?

**Comment by acstll** [comment]
Hey @gaurishkamat 

usually per component, there are 2 files for automated testing:
- `.spec` for unit tests ([Stencil docs](https://stenciljs.com/docs/unit-testing))
- `.e2e` for end-to-end ([Stencil docs](https://stenciljs.com/docs/end-to-end-testing))

I think the `spec` one should be enough for your changes:
https://github.com/telekom/scale/blob/main/packages/components/src/components/text-field/text-field.spec.ts

To run them, you can do either:
- `yarn test --spec` or
- `yarn test --e2e`
within the `packages/components` folder.

It could be also that the snapshots for the component you're working on need to be updated, you do that with `yarn test --spec --updateSnapshot` ([Jest docs](https://jestjs.io/docs/snapshot-testing))

Let me know if I can help with anything else.

**Comment by gaurishkamat** [comment]
Sorry I meant to say, how do I run the component locally.  Just to make sure if my changes reflect. 

Thanks for he explanation!

**Comment by acstll** [comment]
Oh, no problem.

You can create a temporary html file inside [`packages/components/src/html`](https://github.com/telekom/scale/tree/main/packages/components/src/html) (duplicate one of the existing) and then run `yarn dev` from `packages/components`. You will be able to work on that html file in localhost:3000/{file-name}.html

You can delete this html file when you're done so it doesn't end up in the repo.

You could also work directly in Storybook, but you'd have to update it to see your changes (which is part of the task anyways :))
- `yarn build:watch` in `packages/components` in one terminal window
- `yarn start` in `packages/storybook-vue` in another terminal window

If Storybook is too complex, then `yarn dev` should be fine.

**Comment by christopherGdynia** [comment]
Hello there,

can you give me an update, how it is going the datalist? :D
I need this for multiple Projects next months.
If you need anything, I will be there to help

Thanks for your work!

**Comment by acstll** [comment]
There is a PR, but it seems stale #474. Please see my last comments and request for changes: https://github.com/telekom/scale/pull/474#pullrequestreview-725454185.

I think adding the `list` attribute to the `scale-text-field` should be enough. And the user can render the `<datalist>` themselves next to the input. Don't you think?

If you feel like contributing a new PR, please do @christopherGdynia 🙂

@gaurishkamat are you OK with this?

Thank you!

**Comment by gaurishkamat** [comment]
Hi @acstll,
Thank you for reviewing the changes. I will contribute a new PR shortly.

Regards!

**Comment by acstll** [comment]
Closed by #532
